openedx/core/djangoapps/theming/core.py

```diff
@@ -1,62 +1,32 @@
 """
 Core logic for Comprehensive Theming.
 """
-from path import Path
-
+import os.path
+from path import Path as path
 from django.conf import settings
 
+from .helpers import (
+    get_project_root_name,
+)
 
-def comprehensive_theme_changes(theme_dir):
-    """
-    Calculate the set of changes needed to enable a comprehensive theme.
-
-    Arguments:
-        theme_dir (path.path): the full path to the theming directory to use.
-
-    Returns:
-        A dict indicating the changes to make:
-
-            * 'settings': a dictionary of settings names and their new values.
-
-            * 'template_paths': a list of directories to prepend to template
-                lookup path.
 
-    """
-
-    changes = {
-        'settings': {},
-        'template_paths': [],
-    }
-    root = Path(settings.PROJECT_ROOT)
-    if root.name == "":
-        root = root.parent
-
-    component_dir = theme_dir / root.name
-
-    templates_dir = component_dir / "templates"
-    if templates_dir.isdir():
-        changes['template_paths'].append(templates_dir)
-
-    staticfiles_dir = component_dir / "static"
-    if staticfiles_dir.isdir():
-        changes['settings']['STATICFILES_DIRS'] = [staticfiles_dir] + settings.STATICFILES_DIRS
-
-    locale_dir = component_dir / "conf" / "locale"
-    if locale_dir.isdir():
-        changes['settings']['LOCALE_PATHS'] = [locale_dir] + settings.LOCALE_PATHS
-
-    return changes
-
-
-def enable_comprehensive_theme(theme_dir):
+def enable_comprehensive_theming(themes_dir):
     """
     Add directories to relevant paths for comprehensive theming.
+    :param themes_dir: path to base theme directory
     """
-    changes = comprehensive_theme_changes(theme_dir)
+    if isinstance(themes_dir, basestring):
+        themes_dir = path(themes_dir)
+
+    if themes_dir.isdir():
+        settings.DEFAULT_TEMPLATE_ENGINE['DIRS'].insert(0, themes_dir)
+        settings.MAKO_TEMPLATES['main'].insert(0, themes_dir)
+
+    for theme_dir in os.listdir(themes_dir):
+        staticfiles_dir = os.path.join(themes_dir, theme_dir, get_project_root_name(), "static")
```

Comment:
@mattdrayer @ziafazal I think we need to remove `theme_dir` from the line above, otherwise we end up with paths like `/edx/app/edxapp/edx-platform/themes/edx.org/cms/lms/static`.